modules/widgets-panel/assets/js/editor/favorites-component.js

```diff
@@ -0,0 +1,171 @@
+	}
+
+	moveElementInArray( array, from, to ) {
+		array.splice( to, 0, array.splice( from, 1 )[ 0 ] );
+		return array;
+	}
+
+	addToolTip() {
+		const $favoriteCategory = jQuery( '#elementor-panel-category-favorites .elementor-panel-category-title' );
+		if ( 'favorites' === $favoriteCategory.text().trim().toLowerCase() ) {
+			const $iconInfo = jQuery( '<i>', { class: 'eicon-info-circle', 'data-tooltip': __( 'Right click on widgets to add or remove them from favorites', 'elementor' ) } );
+			$iconInfo.appendTo( $favoriteCategory );
+			const icon = $favoriteCategory.find( '> i' );
+			icon.tipsy( {
+				title: function title() {
+					return this.getAttribute( 'data-tooltip' );
+				},
+				gravity: 's',
+			} );
+		}
+	}
+
+	displayFavoriteMenu( action, widgetId, menuText, e ) {
+		// Remove the menu if exists
+		this.removeFavoriteMenu();
+
+		const menuClass = menuText.toLowerCase().replaceAll( ' ', '-' );
+		const $favoriteWidgetMenu = jQuery( '<div>', { id: this.$favoriteMenuId, class: menuClass } );
+		$favoriteWidgetMenu.html( jQuery( '<span>' ).text( menuText ) );
+		$favoriteWidgetMenu.insertAfter( this.$panel );
+
+		this.setPositionMenu( $favoriteWidgetMenu, e );
+
+		if ( widgetId && $favoriteWidgetMenu.length ) {
+			this.contextmenuOnFavoritesMenu();
+			this.contextmenuOrClickOnOutsideMenu();
+
+			$favoriteWidgetMenu.on( 'click.favorites', ( event ) => {
+				event.preventDefault();
+				if ( 'add' === action ) {
+					// Add widget to DB and "FAVORITES" category
+					$e.run( 'panel-favorites/add', { widget: widgetId } );
+				}
+				if ( 'remove' === action ) {
+					// Remove widget from DB and "FAVORITES" category
+					$e.run( 'panel-favorites/remove', { widget: widgetId } );
+				}
+				$favoriteWidgetMenu.off( 'click.favorites', this.removeFavoriteMenu() );
+			} );
+		}
+	}
+
+	setPositionMenu( favoriteWidgetMenu, e ) {
+		const menuWidth = favoriteWidgetMenu.outerWidth();
+		const setMenuPositionX = elementorCommon.config.isRTL ? ( e.pageX - menuWidth ) : e.pageX;
+		if ( favoriteWidgetMenu.length ) {
+			favoriteWidgetMenu.css( {
+				top: e.pageY + 'px',
+				left: setMenuPositionX + 'px',
+			} );
+		}
+	}
+
+	removeFavoriteMenu() {
+		const $favoriteWidgetMenu = jQuery( `#${ this.$favoriteMenuId }` );
+		if ( $favoriteWidgetMenu.length ) {
+			$favoriteWidgetMenu.remove();
+		}
+	}
+
+	addFavoriteWidgetsEvents() {
+		this.contextmenuAddToFavorites();
+		this.contextmenuRemoveFromFavorites();
+	}
+
+	contextmenuAddToFavorites() {
+		const $panelCategories = jQuery( '[ id^=elementor-panel-category- ]:not( [ id=elementor-panel-category-favorites ] )' );
+		if ( $panelCategories.length ) {
+			$panelCategories.on( 'contextmenu', '.elementor-element', ( e ) => {
+				e.preventDefault();
+				const widgetId = e.target.offsetParent.dataset.id;
+				const menuText = __( 'Add To Favorites', 'elementor' );
+				this.displayFavoriteMenu( 'add', widgetId, menuText, e );
+			} );
+		}
+	}
+
+	contextmenuRemoveFromFavorites() {
```

Comment:
Please review the rest of the code according to the comments until here.

Reply:
We talk about it - Fixed